src/adagrams.js

```diff
@@ -1,15 +1,141 @@
+/* eslint-disable quotes */
+const letterPool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
+const scoreChart = {
+  1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+  2: ["D", "G"],
+  3: ["B", "C", "M", "P"],
+  4: ["F", "H", "V", "W", "Y"],
+  5: ["K"],
+  8: ["J", "X"],
+  10: ["Q", "Z"],
+};
+
+////////////////////////////
+// BEGIN HELPER FUNCTIONS //
+////////////////////////////
+
+function getRandomLetter(obj) {
+  const letters = Object.keys(obj);
+  return letters[Math.floor(Math.random() * letters.length) + 1];
+}
+
+function toObject(arr) {
+  const obj = {};
+  for (let i = 0; i < arr.length; i++) {
+    if (arr[i] in obj) {
+      obj[arr[i]] += 1;
+    } else {
+      obj[arr[i]] = 1;
+    }
+  }
+  return obj;
+}
+
+function getKeyByValue(object, value) {
+  return Object.keys(object).find((key) => object[key].includes(value));
+}
+
+const updateBestWord = (obj, word) => {
+  obj.word = word;
+  obj.score = scoreWord(word);
+};
+
+////////////////////////////
+// END HELPER FUNCTIONS //
+////////////////////////////
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  let letterPoolcopy = JSON.parse(JSON.stringify(letterPool));
+  let hand = [];
+  while (hand.length < 10) {
+    let drawnLetter = getRandomLetter(letterPoolcopy);
+    if (letterPoolcopy[drawnLetter] > 0) {
+      hand.push(drawnLetter);
+      letterPoolcopy[drawnLetter] -= 1;
+    }
+  }
+  return hand;
```

Comment:
Overall I see what you were trying to achieve here, just a few small adjustments!

Your function currently forces a false randomness. It will randomly pick one of the 26 letters and remove one,
but it ignores the fact that we have different letters with different frequencies. To get around this, the main
suggestion would be to make a global dictionary like the one you had, and then write a helper function that
converts that dictionary into a list or string that holds every single letter possible. Then we can use that helper
function to create our collection of letters as opposed to making a deep copy! This will also make it so that you don't have tp build a frequency map to make those comparisons!

If we have that collection of all the letters, it makes it really easy to remove letters as you use them!